content/docs/rfcs/51/README.md

```diff
@@ -173,45 +169,103 @@ This example node is part of shards `13`, `14`, and `45` in the Status main-net
 
 # Automatic Sharding
 
-> *Note:* Automatic sharding is not yet part of this specification.
-This section merely serves as an outlook.
-A specification of automatic sharding will be added to this document in a future version.
-
-Automatic sharding is a method for scaling Waku relay in the number of (smaller) content topics.
-It automatically maps Waku content topics to pubsub topics.
-Clients and protocols building on Waku relay only see content topics, while Waku relay internally manages the mapping.
-This provides both scaling as well as removes confusion about content and pubsub topics on the consumer side.
-
-From an app point of view, a subscription to a content topic `waku2/xxx` using automatic sharding would look like:
-
-`subscribe("/waku2/xxx", auto=true)`
-
-The app is oblivious to the pubsub topic layer.
-(Future versions could deprecate the default pubsub topic and remove the necessity for `auto=true`.)
-
-*The basic idea behind automatic sharding*:
-Content topics are mapped using [consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing).
-Like with DHTs, the hash space is split into parts,
-each covered by a Pubsub topic (mesh network) that carries content topics which are mapped into the respective part of the hash space.
+Autosharding selects shards automatically and is the default behavior for shard choice.
+The other choices being static and named sharding as seen in previous sections.
+Shards (pubsub topics) SHOULD be computed from content topics with the procedure below.
+
+## Rendezvous Hashing
+
+Also known as the Highest Random Weight (H.R.W.) method, has many properties useful for shard selection.
+- Distributed agreement. Local only computation without communication.
+- Load balancing. Content topics are spread as equally as possible to shards.
+- Minimal disruption. Content topics switch shards infrequently.
+- Low overhead. Easy to compute.
+
+### Algorithm
+
+For each shard,
```

Comment:
How is the `cluster` value chosen?

Reply:
That question and maybe a redesign of "shard space" should happen in another PR.

Reply:
I agree. We don't have to make all changes in this PR. In fact, shard selection/allocation may be removed completely from this spec and go into a separate "TWN Gen 0" spec in future so we can separate the concepts (stable) from the opinionated implementation in each gen network (unstable, or at least possibly changing for each generation)